applications/crossbar/src/modules/cb_user_auth.erl

```diff
@@ -314,50 +341,73 @@ load_md5_results(Context, JObj) ->
     lager:debug("found MD5 credentials belong to user ~s", [wh_doc:id(JObj)]),
     cb_context:set_doc(Context, wh_json:get_value(<<"value">>, JObj)).
 
-%%--------------------------------------------------------------------
+
+%% @public
+-spec cleanup_reset_ids() -> 'ok'.
+cleanup_reset_ids() ->
+    CreatedBefore = wh_util:current_tstamp() - 2 * ?SECONDS_IN_DAY,
+    ViewOptions = [{'startkey', 0}
+                   ,{'endkey', CreatedBefore}
+                   %% TODO: find a limit that matches the number of pwd resets per day
+                   ,{'limit', couch_util:max_bulk_insert()}
+                   ,'include_docs'
+                  ],
+    case couch_mgr:get_results(?WH_ACCOUNTS_DB, ?LIST_BY_MTIME, ViewOptions) of
```

Comment:
You set `reset_id` on the user though; what happens if their doc is updated by an admin? What if they have a script that sets some property every day, bumping the `pvt_modified` each time (and thus never cleaning up the `reset_id`).

Could it simplify life if you tracked the reset attempt as its own document?
```json
{"_id":"reset_id_256"
 ,"user_id":"user_id_32"
 ,"pvt_created":timestamp
 ,"pvt_type":"password_reset"
}
```

Now you can directly fetch the doc instead of using a view, and looking up the user doc is a `open_cache_doc/2` call away as well.

Cleanup then becomes binding for `crossbar_cleanup:binding_account()` and `cleanup_reset_ids(AccountDb)` can query `maintenance/listing_by_type` with `{'key', <<"password_reset">>}`. I would wager this result set will typically be small enough that iterating over it and checking pvt_modified won't be overly expensive in memory/compute power.

Thoughts?